File: pybind_approach_archive/blocking_test.py
import signal
import time
import sys
import ordlidar

# Handler for signal interruption (SIGINT)
def sig_handle(signo, frame):
    print(f"Program exit, Receive SIGNAL {signo}")
    global running
    running = False
    time.sleep(1)
    sys.exit(1)

# Register the SIGINT handler
signal.signal(signal.SIGINT, sig_handle)

# Main program logic
if __name__ == "__main__":
    running = True

    # Constants representing the lidar type and model
    type = ordlidar.ORADAR_TYPE_SERIAL
    model = ordlidar.ORADAR_MS200

    # Create an instance of OrdlidarDriver from the ordlidar module
    device = ordlidar.OrdlidarDriver(type, model)
    
    # Port name and baudrate configuration
    if sys.platform.startswith('win'):
        port_name = "COM18"
    else:
        port_name = "/dev/ttyACM0"
    serial_baudrate = 230400
    is_logging = True

    count = 0
    device.set_serial_port(port_name, serial_baudrate)

    while running:
        if device.connect():
            print("Lidar device connected successfully.")
            break
        else:
            print(f"Lidar device connection to {port_name} failed.")
            time.sleep(1)

    while running:
        # Assuming the method grab_full_scan_blocking correctly returns a tuple (ret, scan_data)
        ret, scan_data = device.grab_full_scan_blocking(1000)
        if ret:
            count += 1
            print(f"Count = {count}, Point num: {len(scan_data.data)}")
            if is_logging:
                for i, point in enumerate(scan_data.data):
                    print(f"[{i}: Distance={point.distance}, Angle={point.angle}]")
        else:
            print("Error: Failed to get full scan data.")

    device.disconnect()

File: lidar_sdk/samples/blocking_test.cpp
#include "ord_lidar_driver.h"

using namespace std;
using namespace ordlidar;

unsigned char running = 1;
static void sig_handle(int signo)
{
    printf("program exit, [%s,%s] Receive SIGNAL %d ====== \r\n", __FILE__, __func__, signo);
    running = 0;
    delay(1000);
    exit(1);
}

int main(int argc, char *argv[])
{
    (void)argc;(void)argv;
    
    signal(SIGINT, sig_handle);

    uint8_t type = ORADAR_TYPE_SERIAL;
    int model = ORADAR_MS200;

    OrdlidarDriver device(type, model);
    full_scan_data_st scan_data;
    
    #if defined(_WIN32)
    std::string port_name("com18");
    #else
    std::string port_name("/dev/Lidar");
    #endif
    int serialBaudrate = 230400;
    bool is_logging = true;
    bool ret = false;
    long long count = 0;
    device.SetSerialPort(port_name, serialBaudrate);

    while (running)
    {
        if (device.Connect())
        {
            printf("scan_frame_data lidar device connect succuss..\n");
            break;
        }
        else
        {
            printf("lidar device connect %s fail..\n", port_name.c_str());
            delay(1000);
        }
    }

    while (running)
    {
        ret = device.GrabFullScanBlocking(scan_data, 1000);
        if (ret)
        {
            printf("count = %lld, point_num: %d\n", ++count, scan_data.vailtidy_point_num);
            if (is_logging)
            {
                for (int i = 0; i < scan_data.vailtidy_point_num; i++)
                {
                    printf("[%d: %X, %X] \n", i, (scan_data.data[i]), scan_data.data[i].angle);
                }
            }
        }
        else
        {
            printf("error: fail get full scan data\n");
        }
    }

    device.Disconnect();
exit:
    return 0;
}